chemprop/cli/hpopt.py

```diff
@@ -0,0 +1,383 @@
+from chemprop.nn.utils import Activation
+from chemprop.nn import AggregationRegistry
+from chemprop.data import MolGraphDataLoader
+from chemprop.cli.utils.command import Subcommand
+from chemprop.cli.train import (
+    add_train_args,
+    build_datasets,
+    build_model,
+    build_splits,
+    normalize_inputs,
+    process_train_args,
+    validate_train_args,
+)
+from chemprop.cli.common import add_common_args, process_common_args, validate_common_args
+import json
+import logging
+import sys
+from argparse import ArgumentParser, Namespace
+from pathlib import Path
+from copy import deepcopy
+
+import torch
+from lightning import pytorch as pl
+from ray import tune
+from ray.train import CheckpointConfig, RunConfig, ScalingConfig
+from ray.train.lightning import (
+    RayDDPStrategy,
+    RayLightningEnvironment,
+    RayTrainReportCallback,
+    prepare_trainer,
+)
+from ray.train.torch import TorchTrainer
+from ray.tune.schedulers import ASHAScheduler
+
+NO_HYPEROPT = False
+try:
+    from ray.tune.search.hyperopt import HyperOptSearch
+except ImportError:
+    NO_HYPEROPT = True
+
+# NO_OPTUNA = False
+# try:
+#     from ray.tune.search.optuna import OptunaSearch
+# except ImportError:
+#     NO_OPTUNA = True
+
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_SEARCH_SPACE = {
```

Comment:
If you have this dict as a global variable it will be created at import time of this module. This dict requires that the hpopt packages are installed. You can fix  this by:

 - moving this defintion into the try/except block where you do the imports
 - inside the CLI main class, move the import of the hopt subcommand into the body of the cli function to avoid importing until needed

The former is easier